lib/elixir/lib/io.ex

```diff
@@ -171,14 +178,13 @@ defmodule IO do
   end
 
   @doc """
-  Writes the argument to the device, similar to `write/2`,
-  but adds a newline at the end. The argument is expected
-  to be a chardata.
+  Writes the given argument to the `device`, similar to `write/2`,
+  but adds a newline at the end.
+
   """
   @spec puts(device, chardata | String.Chars.t) :: :ok
   def puts(device \\ group_leader(), item) do
-    erl_dev = map_dev(device)
```

Comment:
This variable serves as some kind of documentation IMO; I'm not sure if we should refactor it out (or if that change belongs to a style/docs PR).